src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/modelchecking/ModelCheckingStrategy.kt

```diff
@@ -260,14 +286,14 @@ internal class ModelCheckingStrategy(
     ) {
         private var lastNotInitializedNode: SwitchChoosingNode? = initialLastNotInitializedNode
         private lateinit var interleavingFinishingRandom: Random
-        private lateinit var nextThreadToSwitch: Iterator<Int>
+        private var currentInterleavingPosition = 0
```

Comment:
Would it make sense to initialize to `-1` instead?
`0` is technically a valid position in a list.

Reply:
this would only change the way nextThread is returned in chooseThread method: prefix ++ instead of suffix one.
I think zero as initial value is more understandable